### ISSUE TITLE
MySQL BigInt Unsigned

### ISSUE THREAD

**Issue by hatemjaber** [drizzle-team/drizzle-kit-mirror#22]
I'm trying to generate a field that looks like this:

`id bigint(20) unsigned NOT NULL`

using this in my drizzle schema:

`id: bigint('id', { mode: 'bigint' }).primaryKey().notNull()`

I looked through the code and the examples but can't get it to generate an unsigned bigint. At this point I'm not sure if it's supported or not, I'm using the latest version of drizzle.

The other thing that I can't seem to figure out is how to run the migration, I tried:

`npx drizzle-kit up:mysql`

Thank you in advance for your help!

**Comment by hatemjaber** [comment]
Just read in the docs we can create a custom type and that's solves the problem for the bigint unsigned:

```
const bigintUnsiged = customType<{ data: bigint; }>({
    dataType() {
        return 'bigint(20) unsigned';
    }
});
```

The only thing that I'm not sure if it's right is the following: 

```
import { migrate } from 'drizzle-orm/mysql2/migrator';
import { db } from ".";

(async () => await migrate(db, { migrationsFolder: './.migrations' }))();
```

Are we suppose to the run the migrations like this? By calling some script that runs the `migrate()` function?

**Comment by AndriiSherman** [comment]
@hatemjaber wow, great usage for customTypes!
`bigint unsigned` is in backlog, so currently yeah, you can just use customType
You can also provide `20` as param(I guess we have some examples in docs for this case)

Regarding migration - there are 2 options for now. First is migration script, that could be inside you backend(that is invoked on app starts) or custom resource and separate script, that is invoked on your code deploy

Also for mysql we have `db push` feature, but it's still in internal release. If you wan to give it a try - here are some [docs](https://github.com/drizzle-team/drizzle-orm/blob/db-push-docs/docs/db-push.preview.md)

**Comment by hatemjaber** [comment]
@AndriiSherman thank you for replying. I tried to pass 20 as a param but I couldn't find it in the documentation. I was actually going to suggest that length and signed/unsigned be options for bigint to round it off and make it more complete.

I am looking forward to the db push, that's going to be a nice feature. I will just run it on the script that I made for now until you guys make it available.
